futures-core/src/stream.rs

```diff
@@ -126,7 +165,7 @@ mod private_try_stream {
 
 /// A convenience for streams that return `Result` values that includes
 /// a variety of adapters tailored to such futures.
-pub trait TryStream: private_try_stream::Sealed {
+pub trait TryStream: Stream + private_try_stream::Sealed {
```

Comment:
Added this bound because it would be convenient if `stream.size_hint()` could be called on `St: TryStream`.

Reply:
oh...
```text
error[E0191]: the value of the associated type `Item` (from the trait `futures_core::stream::Stream`) must be specified
  --> futures/tests/object_safety.rs:20:30
   |
20 |     assert_is_object_safe::<&dyn TryStream<Ok = (), Error = ()>>();
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ associated type `Item` must be specified
```